test/engine-event.test.js

```diff
@@ -301,6 +301,69 @@ describe('Engine: event', () => {
     })
   })
 
+  context('engine events: with facts', () => {
```

Comment:
how about a test that provides the other supported properties/features - namely `params:` and `path:`